python_modules/dagster/dagster/_core/storage/event_log/sql_event_log.py

```diff
@@ -132,6 +133,7 @@ def prepare_insert_event(self, event):
             partition=partition,
         )
 
+    @functools.lru_cache(maxsize=None)
```

Comment:
this is a kinda funny one -- we call this a bunch in the get_asset_records code, and it requires querying the database each time. This added up to around 10 seconds of work in one test!

I figured the result of this wouldn't change from call to call, so why not cache it? maxsize=None because there's not a lot of distinct inputs to this function and it's just returning a bool.

Reply:
Can we test that this call will still function correctly if someone run a migration between calls to this method?

We could make another call here to test that `instance.can_cache_asset_status_data()` will return `True` after the migration runs: https://github.com/dagster-io/dagster/blob/a54d17030f4cb9aab20b5498dc23f31913636b5c/python_modules/libraries/dagster-postgres/dagster_postgres_tests/compat_tests/test_back_compat.py#L776

If this test fails, we might need to come up with caching within `get_asset_records`

Reply:
that's a good point, I'd imagine it would fail -- i'll update get_asset_records directly instead